group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinator.java

```diff
@@ -170,29 +170,27 @@ CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection> delet
      * Fetch offsets for a given Group.
      *
      * @param context           The request context.
-     * @param groupId           The group id.
-     * @param topics            The topics to fetch the offsets for.
+     * @param request           The OffsetFetchRequestGroup request.
      *
      * @return A future yielding the results or an exception.
      */
-    CompletableFuture<List<OffsetFetchResponseData.OffsetFetchResponseTopics>> fetchOffsets(
+    CompletableFuture<OffsetFetchResponseData.OffsetFetchResponseGroup> fetchOffsets(
         RequestContext context,
-        String groupId,
-        List<OffsetFetchRequestData.OffsetFetchRequestTopics> topics,
+        OffsetFetchRequestData.OffsetFetchRequestGroup request,
         boolean requireStable
     );
 
     /**
      * Fetch all offsets for a given Group.
      *
      * @param context           The request context.
-     * @param groupId           The group id.
+     * @param request           The OffsetFetchRequestGroup request.
      *
      * @return A future yielding the results or an exception.
      */
-    CompletableFuture<List<OffsetFetchResponseData.OffsetFetchResponseTopics>> fetchAllOffsets(
+    CompletableFuture<OffsetFetchResponseData.OffsetFetchResponseGroup> fetchAllOffsets(
```

Comment:
Does it make sense to merge the fetchOffsets and fetchAllOffsets which have the same arguments and outputs format? We can determine whether fetches all offsets by checking the provided topics

Reply:
That's a good question. I personally find it a little clearer as it is but I don't feel very strong about it. @jolshan What do you think? Anyway, I would consider doing this separately as it requires more changes.

Reply:
Given we now pass the entire request I think it makes more sense than before. However, I don't feel strongly about either way.